### PR TITLE
Change vocation map to unordered_set and improve weapons.cpp code format

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Improve code format

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by EPuncker — DISMISSED** [review]

File: src/weapons.cpp [hunk 1]
```diff
@@ -262,60 +262,86 @@ int32_t Weapon::playerWeaponCheck(Player* player, Creature* target, uint8_t shoo
...
+
+	if (player->getLevel() < getReqLevel()) {
+		damageModifier = (isWieldedUnproperly() ? damageModifier / 2 : 0);
 	}
 
-	return 100;
+	if (player->getMagicLevel() < getReqMagLv()) {
+		damageModifier = (isWieldedUnproperly() ? damageModifier / 2 : 0);
+	}
+	return damageModifier;
```
Comment (ranisalt) [inline src/weapons.cpp:304]: Newline above this line

File: src/weapons.cpp [hunk 2]
```diff
@@ -262,60 +262,86 @@ int32_t Weapon::playerWeaponCheck(Player* player, Creature* target, uint8_t shoo
...
+	}
+
+	if (player->getSoul() < soul) {
+		return false;
+	}
+
+	if (!vocWeaponMap.empty()) {
+		auto it = vocWeaponMap.find(player->getVocationId());
+		if (it == vocWeaponMap.end()) {
+			return false;
```
Comment (ranisalt) [inline src/weapons.cpp:344]: While at that, can you replace this map with an unordered set? We have decent containers to use, this is not Go.

**Review by ranisalt — CHANGES_REQUESTED** [review]

File: src/spells.cpp [hunk 3]
```diff
@@ -510,7 +509,9 @@ bool Spell::configureSpell(const pugi::xml_node& node)
 		int32_t vocationId = g_vocations.getVocationId(attr.as_string());
 		if (vocationId != -1) {
 			attr = vocationNode.attribute("showInDescription");
-			vocSpellMap[vocationId] = !attr || attr.as_bool();
+			if (attr && attr.as_bool()) {
+				vocationSpellSet.insert(vocationId);
+			}
```
Comment (ramon-bernardo) [inline src/spells.cpp:514]: It's not correct, will revert

File: src/movement.h [hunk 4]
```diff
@@ -119,14 +119,18 @@ class MoveEvent final : public Event
 	const std::string& getVocationString() const { return vocationString; }
 	void setVocationString(const std::string& str) { vocationString = str; }
 	uint32_t getWieldInfo() const { return wieldInfo; }
-	const VocEquipMap& getVocEquipMap() const { return vocEquipMap; }
-	void addVocEquipMap(std::string vocName)
+	const VocationEquipSet& getVocationEquipSet() const { return vocationEquipSet; }
```
Comment (ranisalt) [inline src/movement.h:122]: What do you think about returning `const auto&` and getting rid of the alias? One less thing polluting the global namespace.

File: src/movement.h [hunk 5]
```diff
@@ -119,14 +119,18 @@ class MoveEvent final : public Event
...
+	void addVocationEquipSet(std::string vocationName)
```
Comment (ranisalt) [inline src/movement.h:123]:
```suggestion
	void addVocationEquipSet(const std::string& vocationName)
```

File: src/spells.h [hunk 6]
```diff
@@ -131,8 +134,18 @@ class Spell : public BaseSpell
 	bool isLearnable() const { return learnable; }
 	void setLearnable(bool l) { learnable = l; }
 
-	const VocSpellMap& getVocMap() const { return vocSpellMap; }
-	void addVocMap(uint16_t n, bool b) { vocSpellMap[n] = b; }
+	const VocationSpellMap& getVocationSpellMap() const { return vocationSpellMap; }
+	void addVocationSpellMap(std::string vocationName, bool showInDescription)
```
Comment (ranisalt) [inline src/spells.h:138]:
```suggestion
	void addVocationSpellMap(const std::string& vocationName, bool showInDescription)
```

File: src/weapons.h [hunk 7]
```diff
@@ -105,21 +107,26 @@ class Weapon : public Event
 	uint32_t getWieldInfo() const { return wieldInfo; }
 	void setWieldInfo(uint32_t info) { wieldInfo |= info; }
 
-	void addVocWeaponMap(std::string vocName)
+	const VocationWeaponSet& getVocationWeaponSet() const { return vocationWeaponSet; }
+	void addVocationWeaponSet(std::string vocationName)
```
Comment (ranisalt) [inline src/weapons.h:111]:
```suggestion
	void addVocationWeaponSet(const std::string& vocationName)
```

**Review by ranisalt — CHANGES_REQUESTED** [review]
Just some avoidable copies

**Review by EPuncker — APPROVED** [review]

File: src/spells.h [hunk 8]
```diff
@@ -186,7 +185,7 @@ class Spell : public BaseSpell
 	bool playerInstantSpellCheck(Player* player, const Position& toPos);
 	bool playerRuneSpellCheck(Player* player, const Position& toPos);
 
-	VocationSpellMap vocationSpellMap;
+	std::map<uint16_t, bool> vocationSpellMap;
```
[the remark this answers is not in the document]
Comment (ramon-bernardo) [reply src/spells.h:188]: No, the value of map is used to show in description

**Review by EPuncker — APPROVED** [review]

File: src/weapons.h (hunk 7 above)
Comment (ramon-bernardo) [reply src/weapons.h]: Thanks!

File: src/weapons.cpp (hunk 2 above)
Comment (ramon-bernardo) [reply src/weapons.cpp]: Thanks!

**Review by ranisalt — APPROVED** [review]
LGTM

**Pull request merged**